app/src/main/java/de/westnordost/streetcomplete/osm/surface/Surface.kt

```diff
@@ -0,0 +1,85 @@
+package de.westnordost.streetcomplete.osm.surface
+
+import de.westnordost.streetcomplete.osm.surface.Surface.*
+
+enum class Surface(val osmValue: String) {
+    ASPHALT("asphalt"),
+    CONCRETE("concrete"),
+    CONCRETE_PLATES("concrete:plates"),
+    CONCRETE_LANES("concrete:lanes"),
+    FINE_GRAVEL("fine_gravel"),
+    PAVING_STONES("paving_stones"),
+    COMPACTED("compacted"),
+    DIRT("dirt"),
+    SETT("sett"),
+    UNHEWN_COBBLESTONE("unhewn_cobblestone"),
+    GRASS_PAVER("grass_paver"),
+    WOOD("wood"),
+    WOODCHIPS("woodchips"),
+    METAL("metal"),
+    GRAVEL("gravel"),
+    PEBBLES("pebblestone"),
+    GRASS("grass"),
+    SAND("sand"),
+    ROCK("rock"),
+    CLAY("clay"),
+    ARTIFICIAL_TURF("artificial_turf"),
+    TARTAN("tartan"),
+    PAVED_ROAD("paved"),
+    UNPAVED_ROAD("unpaved"),
+    GROUND_ROAD("ground"),
+    PAVED_AREA("paved"),
+    UNPAVED_AREA("unpaved"),
+    GROUND_AREA("ground"),
+
+    // extra values, recording duplicates
+    MUD("mud"), // valid value, but not displayed specially in StreetComplete
+    EARTH("earth"), // synonym of "dirt", maybe more clear
+    CHIPSEAL("chipseal"), // subtype/synonym of asphalt
+    METAL_GRID("metal_grid"), // not exactly the same info, but better to show it
+    // this way than without info
+
+    // this values ideally would be removed from OpenStreetMap, but while they remain
+    // we want to handle them somehow
+    SOIL("soil"),
+    PAVING_STONES_WITH_WEIRD_SUFFIX("paving_stones:30"),
+    COBLLESTONE_FLATTENED("cobblestone:flattened"),
+    BRICK("brick"),
+    BRICKS("bricks"),
+    UNIDENTIFIED("fake_value");
```

Comment:
I experimented with `Surface` and `UnidentifiedSurface` being children of `SurfaceMaybeUnidentifiedOne` but it made things extremely ugly.

It seems that special handling in `SurfaceAndNote.applyTo` results in better effects but maybe there is a better way to handle this.

Reply:
Yeah... 

```kotlin

sealed interface ParsedSurface
enum class Surface : ParsedSurface
object UnknownSurface : ParsedSurface`

```
(or with a different name instead of `ParsedSurface`) would definitely be cleaner. What exactly does it maker more ugly? It seems to be, it is only making things ugly in the sense that null safety makes things "ugly" because the language forces you to check for null.

Reply:
redid it in a smarter way and seems OK this time - offloaded some class-dependent stuff into helper functions